docs/offline-support.adoc

```diff
@@ -0,0 +1,218 @@
+ifdef::env-github[]
+// If on GitHub, define attributes so we can find our diagram files and render
+// them.
+
+// The branch will be used to find the correct diagrams to render below.
+// When PRing changes to the diagrams you can change this attributes
+// temporarily to the name of the branch you're working on. But don't forget
+// to change it back to main before merging!!
+:github-branch: add-offline-docs
+
+:github-repo: nhsuk/record-childrens-vaccinations
+
+// URL for PlantUML Proxy. Using an attribute mainly because it's just tidier.
+:plantuml-proxy-url: http://www.plantuml.com/plantuml/proxy?cache=no&src=
+
+// Full path prefix we'll use for diagrams below.
+:diagram-path-url: {plantuml-proxy-url}https://raw.githubusercontent.com/{github-repo}/{github-branch}/docs
+endif::[]
+
+:toc:
+
+= Offline support
+
+A key feature required for the Record children's vaccines service is to allow
+clinical users to use parts of the system while not connected to the Internet.
+There are certain clinical settings where Internet access isn't available, for
+example in schools where there may be no mobile signal or WiFi that the SAIS
+team can use.
+
+To accomplish this the standard patterns provided by server-rendered apps are
+insufficient. Aggressive caching can help accommodate viewing data, but updating
+records requires a more advanced approach.
+
+== Feature overview
+
+The current design of the service expects the SAIS team to save a campaign for
+offline use in advance using a button in the UI. They will then be able to
+retrieve and view saved patient information, including consent status and other
+info (e.g. past vaccination history), before performing the vaccination. When
+the vaccination is performed, the event will be recorded to their local system.
+Once reconnected to the Internet, the records will be uploaded from the local
+system to the service.
+
+== Technical approach
+
+IMPORTANT: The design of this service feature is still being designed and tested
+           with users. The exact approach described here may change, e.g. the
+           user may need to manually initiate the synchronisation back to the
+           server, etc.
+
+=== Service Worker
+
+The service worker, a feature widely available in modern browsers, is used to
```

Comment:
> feature widely available in modern browsers

worth linking to https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker#browser_compatibility here? I was wondering which browsers were supported so I went here myself...

Reply:
Good call. Added link.